### PR TITLE
Change recipe sort to difficulty, name, crafting time

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Presently the recipes in the crafting menu are sorted by their difficulty factor, which makes the sort order seem completely random, seeing how the factor isn't exposed to the player at all. That would've made some sense had the only way to learn recipes been by practicing the relevant skills, but this is not the case. This PR changes the sort order to alphabetical, making it easier to keep track of what items you can craft.
Implements changes outlined in #47994 

#### Describe the solution
Sort on recipe difficulty first, but when those are the same sort on name, then as a final tie breaker sort on crafting time.

#### Describe alternatives you've considered
eh

#### Testing
Entered crafting screen with many recipes unlocked and navigated around a bit to verify that recipes were sorting as expected.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
> Sort on recipe difficulty first, but when those are the same sort on name, then as a final tie breaker sort on crafting time.

It sounded like you intended to sort by name first, then difficulty? Given that you just explained that sorting by difficulty appears arbitrary

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -816,8 +816,15 @@ const recipe *select_crafting_recipe( int &batch_size_out )
...
-                        return b->difficulty < a->difficulty;
+                    [&player_character]( const recipe * a, const recipe * b ) {
+                        if( b->difficulty != a->difficulty ) {
+                            return b->difficulty < a->difficulty;
+                        }
+                        if( a->result_name() != b->result_name() ) {
+                            return localized_compare( a->result_name(), b->result_name() );
+                        }
+                        return a->time_to_craft( player_character ) <
+                               a->time_to_craft( player_character );
```
Comment (Saicchi) [inline src/crafting_gui.cpp:827]:
```suggestion
                        return b->time_to_craft( player_character ) <
                               a->time_to_craft( player_character );
```

**Comment by MarlossCDDA** [comment]
> 
> 
> > Sort on recipe difficulty first, but when those are the same sort on name, then as a final tie breaker sort on crafting time.
> 
> It sounded like you intended to sort by name first, then difficulty? Given that you just explained that sorting by difficulty appears arbitrary

As discussed on the earlier PR https://github.com/CleverRaven/Cataclysm-DDA/pull/47994#issuecomment-797200455, I consider the fact that difficulty is the first thing we sort on to be a feature rather than a bug - it helps a lot when you're using crafting to train your skills. Definitely agree that within a given difficulty level we should be sorting by name rather than arbitrarily.

File: src/crafting_gui.cpp (hunk 1 above)
Comment (kevingranade) [reply src/crafting_gui.cpp:827]: Whoops. thanks.

**Comment by kevingranade** [comment]
Everything sorted by difficulty ends up fairly incoherent, but grouping by difficulty and then sorting by name ends up pretty good.

**Pull request merged**